push-and-sign-artifact/action.yaml

```diff
@@ -0,0 +1,91 @@
+name: "Push and sign an artifact"
+author: "Robert Schardt <robert.schardt@greenbone.net>"
+description: "An action that pushes an artifact with oras and signs it with cosign"
+
+inputs:
+  product:
+    type: choice
+    description: "The product to build."
+    options:
+      - appliance
+      - asset
+      - lookout
+      - sensor
+      - universal
+      - management-console
+    required: true
+  artifact-url:
+    type: string
+    description: "The artifact-url to upload the artifact into (with tag)."
+    required: true
+  artifact-folder:
+    type: string
+    description: "The path to the system folder to save the artifact file. Default is /tmp."
+    default: '/tmp'
+  registry-user:
+    type: string
+    description: "Registry user name."
+  registry-domain:
+    type: string
+    description: "Registry domain."
+  registry-token:
+    type: string
+    description: "Registry user token. Set is input only if you want to upload an artifact."
+  cosign-key:
+    type: string
+    description: "Key for Signing artifacts/containers."
+    required: true
+  cosign-password:
+    type: string
+    description: "Password for COSIGN key."
+    required: true
+
+branding:
+  icon: "package"
+  color: "green"
+
+runs:
+  using: "composite"
+  steps:
+      - name: Setup oras
+        if: inputs.registry-token
+        shell: bash
+        run: |
+          VERSION="1.2.0"
+          curl -LO "https://github.com/oras-project/oras/releases/download/v${VERSION}/oras_${VERSION}_linux_amd64.tar.gz"
```

Comment:
Please validate the downloaded artifact against a hashsum. Attackers could upload a different tarfile without noticing.